sonic-net/sonic-buildimage
# [dhcp_server] Add dhcp_server container

#### Why I did it
Add dhcp_server ipv4 feature to SONiC.
HLD: https://github.com/sonic-net/SONiC/pull/1282

##### Work item tracking
- Microsoft ADO **(number only)**: 24978865

#### How I did it
To be clarify: This container is disabled by `INCLUDE_DHCP_SERVER = n` for now, which would cause container not build.

1. Add `INCLUDE_DHCP_SERVER` to indicate whether to build dhcp_server container
2. Add docker file for dhcp_server, build and install kea-dhcp4 inside container
3. Add template file for dhcp_server container services.
4. Add entry for dhcp_server to `FEATURE` table in config_db.


#### How to verify it
Build image with `INCLUDE_DHCP_SERVER = y` to verify:
1. Image can be install successfully without crush.
2. By `config feature state dhcp_server enabled` to enable dhcp_server.




#### Description for the changelog


#### Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
